code/core/system/proxy-camera/src/ProxyCamera.cpp

```diff
@@ -32,25 +43,99 @@ namespace proxy {
 
 using namespace std;
 using namespace odcore::base;
+using namespace odcore::data;
+using namespace odtools::recorder;
 
 ProxyCamera::ProxyCamera(const int &argc, char **argv)
-    : TimeTriggeredConferenceClientModule(argc, argv, "proxy-camera") {}
+    : TimeTriggeredConferenceClientModule(argc, argv, "proxy-camera"),
+    m_recorder(),
+    m_camera() {}
 
 ProxyCamera::~ProxyCamera() {}
 
-void ProxyCamera::setUp() {}
+void ProxyCamera::setUp() {
+            // This method will be call automatically _before_ running body().
+            if (getFrequency() < 20) {
+                cerr << endl << endl << "proxy-camera: WARNING! Running proxy-camera with a LOW frequency (consequence: data updates are too seldom and will influence your algorithms in a negative manner!) --> suggestions: --freq=20 or higher! Current frequency: " << getFrequency() << " Hz." << endl << endl << endl;
+            }
+
+            // Get configuration data.
+            KeyValueConfiguration kv = getKeyValueConfiguration();
+
+            // Create built-in recorder.
+            const bool useRecorder = kv.getValue<uint32_t>("proxy-camera.useRecorder") == 1;
+            if (useRecorder) {
+                // URL for storing containers.
+                stringstream recordingURL;
+                recordingURL << "file://" << "proxy_camera" << TimeStamp().getYYYYMMDD_HHMMSS() << ".rec";
```

Comment:
Should I define "odcore::data::TimeStamp m_startOfRecording” in ProxyCamera.h  and then do something similar below in the cpp file to set the format YYYYMMDD_HHMMSS:
m_startOfRecording = TimeStamp();
m_startOfRecording.getYYYYMMDD_HHMMSS()